js/pom.xml

```
@@ -27,6 +27,6 @@
         <!-- The JavaScript projects use the non-standard 'src' folder for their sources, therefore, name it here explicitly -->
         <maven.build.cache.input.1>src</maven.build.cache.input.1>
         <!-- The child projects will package from the 'dist' folder, which is listed as a resource, but isn't a source folder -->
-        <maven.build.cache.exclude.value.1>${project.basedir}/dist</maven.build.cache.exclude.value.1>
+        <maven.build.cache.exclude.value.1>dist</maven.build.cache.exclude.value.1>
```

I took a quick look at the compiled output of the various JavaScript projects, and the following directories containing compiled code are not actively ignored:

- `js/apps/account-ui/lib`
- `js/libs/keycloak-admin-client/lib`

Perhaps we should be ignoring the `lib` directory in it's entirety as well?

```suggestion
        <maven.build.cache.exclude.value.1>dist</maven.build.cache.exclude.value.1>
        <maven.build.cache.exclude.value.2>lib</maven.build.cache.exclude.value.2>
```

I am not too familiar with how the build cache extension works. Why does the `dist` directory "listed as a resource"? What makes it qualify for that?

---

In the `keycloak-account-ui`'s pom.xml, you find the following snippet: 

```
<resources>
            <resource>
                <directory>maven-resources</directory>
            </resource>
            <resource>
                <directory>dist</directory>
                <targetPath>theme/keycloak.v3/account/resources</targetPath>
                <excludes>
                    <exclude>index.html</exclude>
                </excludes>
            </resource>
        </resources>
```

The `dist` folder for the account ui contains only generated content. For the maven resource plugin to copy it to the target folder, it is added as above. 

Adding it as a resource makes the cache extension treat it like a source folder. But it needs to be ignored by the maven cache extension, as those are not sources, but generated files. That's why it is added.  

I don't know if the JavaScript build could put those items directly in the `target` folder which would simplify the Maven configuration. Putting it into the target folder would then also use Maven's clean which might be a good or a bad thing (I wouldn't know).

---

> I don't know if the JavaScript build could put those items directly in the target folder which would simplify the Maven configuration.

Yes, this can be done, it would be a small configuration tweak to accomplish.

> Putting it into the target folder would then also use Maven's clean which might be a good or a bad thing (I wouldn't know).

That would be no issue on the JavaScript side, but likewise the directory could be cleared by JavaScript tooling. If that is not an issue for Maven then I could add some code to this PR, or raise a separate PR, whatever you prefer.

---

+1 for a separate PR. Should this one be merge first, or do you want to have your PR go in first (and then I adjust this one to make it smaller)?

---

Let's get this one merged first, I'll base my work on top of it.

---

Put a [draft PR ](https://github.com/keycloak/keycloak/pull/30490) up which we can rebase and merge after this PR is merged.